Scripts/Extension/GameObjectExtensions.cs

```diff
@@ -29,5 +29,34 @@ public static void TrySetActive(this GameObject gameObject, bool state)
                 gameObject.SetActive(state);
             }
         }
+
+        /// <summary>
+        /// Attempts to retreive the <see cref="Rigidbody"/> or if one is not found then optionally search children then search parents.
+        /// </summary>
+        /// <param name="gameObject">The reference <see cref="GameObject"/> to search on.</param>
+        /// <param name="searchAncestors">Optionally searches all ancestors in the hierarchy for a rigidbody.</param>
+        /// <param name="searchDescendants">Optionally searches all descendants in the hierarchy for a rigidbody.</param>
+        /// <returns>The found <see cref="Rigidbody"/></returns>
+        public static Rigidbody FindRigidbody(this GameObject gameObject, bool searchDescendants = true, bool searchAncestors = false)

```

Comment:
Change this to be generic instead:
```c#
public static T FindComponent<T>(this GameObject gameObject, bool searchDescendants = true, bool searchAncestors = false) where T : Component
```
Update the documentation, tests, type names and usages accordingly.

**This should probably be replacing the `TryGetComponent` method then as it's a more powerful replacement. Beware of the default parameter values, though, as the `TryGetComponent` method didn't act like that so far.**